static/app/views/issueList/groupSearchViewTabs/draggableTabBar.spec.tsx

```diff
@@ -0,0 +1,378 @@
+    {key: '3', label: 'Regressed', content: <div>Tab 3 Content</div>},
+  ];
+
+  describe('Tabs render as expected', () => {
+    it('should render tabs from props', () => {
+      render(
+        <DraggableTabBar
+          tabs={tabs}
+          setTabs={jest.fn()}
+          onTabRenamed={mockOnTabRenamed}
+          showTempTab={false}
+          tempTab={tempTab}
+          defaultNewTab={defaultNewTab}
+        />
+      );
+      expect(screen.getAllByRole('tab').length).toBe(tabs.length);
+      // The query count is included in the tab name here
+      expect(screen.getByRole('tab', {name: 'Prioritized 20'})).toBeInTheDocument();
+      expect(screen.getByRole('tab', {name: 'For Review 1000+'})).toBeInTheDocument();
+      expect(screen.getByRole('tab', {name: 'Regressed'})).toBeInTheDocument();
+
+      expect(screen.getByRole('tabpanel')).toHaveTextContent('Tab 1 Content');
+    });
+
+    it('should render temp tab if showTempTab = true', () => {
+      render(
+        <DraggableTabBar
+          tabs={tabs}
+          setTabs={jest.fn()}
+          onTabRenamed={mockOnTabRenamed}
+          showTempTab
+          tempTab={tempTab}
+          defaultNewTab={defaultNewTab}
+        />
+      );
+      expect(screen.getByRole('tab', {name: 'Unsaved'})).toBeInTheDocument();
+    });
+  });
+
+  describe('Tab menu options', () => {
+    it('should render the correct set of actions for changed tabs', async () => {
+      render(
+        <DraggableTabBar
+          tabs={tabs}
+          setTabs={jest.fn()}
+          onTabRenamed={mockOnTabRenamed}
+          showTempTab
+          tempTab={tempTab}
+          defaultNewTab={defaultNewTab}
+        />
+      );
+      // const prioritzedTabOptions =
+      await userEvent.click(
+        screen.getByRole('button', {name: 'Prioritized Tab Options'})
+      );
+
+      expect(await screen.findByText('Save Changes')).toBeInTheDocument();
+      expect(await screen.findByText('Discard Changes')).toBeInTheDocument();
+      expect(await screen.findByText('Rename')).toBeInTheDocument();
+      expect(await screen.findByText('Duplicate')).toBeInTheDocument();
+      expect(await screen.findByText('Delete')).toBeInTheDocument();
+    });
+
+    it('should render the correct set of actions for unchanged tabs', async () => {
+      render(
+        <DraggableTabBar
+          tabs={tabs}
+          setTabs={jest.fn()}
+          onTabRenamed={mockOnTabRenamed}
+          showTempTab
+          tempTab={tempTab}
+          defaultNewTab={defaultNewTab}
+        />
+      );
+      // We need to explicitly click on the For Review tab since it is not the default (first) tab in props
+      await userEvent.click(screen.getByRole('tab', {name: 'For Review 1000+'}));
+      await userEvent.click(
+        await screen.findByRole('button', {name: 'For Review Tab Options'})
+      );
+
+      expect(await screen.findByText('Rename')).toBeInTheDocument();
```

Comment:
For asserting on these menu items, it would be preferred to query by role. I believe the role of the menu items are `menuitemradio`